warehouse/packaging/models.py

```diff
@@ -142,8 +142,9 @@ def __getitem__(self, version):
             return (
                 session.query(Release)
                 .filter(
-                    (Release.project == self)
-                    & (Release.canonical_version == canonical_version)
+                    Release.project == self,
+                    Release.yanked.is_(False),
```

Comment:
Do we want to do this? This will make it so yanked versions can't be viewed online at all right?